File: csharp/allergies/Allergies.cs
using System;
using System.Collections.Generic;

[Flags]
public enum Allergen
{
    Eggs = 1,
    Peanuts = 2,
    Shellfish = 4,
    Strawberries = 8,
    Tomatoes = 16,
    Chocolate = 32,
    Pollen = 64,
    Cats = 128
}

public class Allergies
{
    public int mask;
    public Allergies(int mask) => this.mask = mask;
    
    public bool IsAllergicTo(Allergen allergen) 
    {
        return (mask & (int)allergen) == (int) allergen;
    }
    public Allergen[] List()
    {
        var result = new List<Allergen>();
        foreach(Allergen x in Enum.GetValues(typeof(Allergen)))
        {
            if(IsAllergicTo(x))
            {
               result.Add(x); 
            }
        }
        return result.ToArray();
    }

}

File: javascript/allergies/allergies.js
export class Allergies {
  constructor(allergiesMask) {
    this.allergiesMask = allergiesMask;
  }

  list() {
    return [...allergies.keys()].filter(allergy => this.allergicTo(allergy));
  }

  allergicTo(allergen) {
    return (allergies.get(allergen) & this.allergiesMask) === allergies.get(allergen);
  }
}

const allergies = new Map([
  [ 'eggs', 1 ],
  [ 'peanuts', 2 ],
  [ 'shellfish', 4 ],
  [ 'strawberries', 8 ],
  [ 'tomatoes', 16 ],
  [ 'chocolate', 32 ],
  [ 'pollen', 64 ],
  [ 'cats', 128 ]
]);